diskqueue.go

```diff
@@ -394,6 +432,17 @@ func (d *diskQueue) writeOne(data []byte) error {
 		return err
 	}
 
+	totalBytes := int64(4 + dataLen)
+
+	// check if we reached the file size limit with this message
+	if d.diskLimitFeatIsOn && d.writePos+totalBytes+8 >= d.maxBytesPerFile {
+		// write number of messages in binary to file
+		err = binary.Write(&d.writeBuf, binary.BigEndian, d.writeMessages)
```

Comment:
a question rather than comment:  does `d.writeMessages` reflect the correct total number here or off by 1? asking because it started with 0 and only incremented later when the file is not finished.

Reply:
Nice catch! writeMessages would be off by 1.